include/nlohmann/detail/macro_scope.hpp

```diff
@@ -395,7 +395,7 @@
 
 #define NLOHMANN_JSON_TO(v1) nlohmann_json_j[#v1] = nlohmann_json_t.v1;
 #define NLOHMANN_JSON_FROM(v1) nlohmann_json_j.at(#v1).get_to(nlohmann_json_t.v1);
-#define NLOHMANN_JSON_FROM_WITH_DEFAULT(v1) nlohmann_json_t.v1 = nlohmann_json_j.value(#v1, nlohmann_json_default_obj.v1);
+#define NLOHMANN_JSON_FROM_WITH_DEFAULT(v1) nlohmann_json_t.v1 = !nlohmann_json_j.is_null() ? nlohmann_json_j.value(#v1, nlohmann_json_default_obj.v1) : nlohmann_json_default_obj.v1;
```

Comment:
Can you please also adjust `NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT` to keep the symmetry between the macros?

Reply:
What change are you expecting to see there?  `NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT` uses `NLOHMANN_JSON_FROM_WITH_DEFAULT`.

Reply:
Oh, missed that. Sorry.

Reply:
I don't understand what you are asking

`NLOHMANN_JSON_FROM_WITH_DEFAULT` is called by all the following macros:

- `NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT`
- `NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT`
- `NLOHMANN_DEFINE_DERIVED_TYPE_INTRUSIVE_WITH_DEFAULT`
- `NLOHMANN_DEFINE_DERIVED_TYPE_NON_INTRUSIVE_WITH_DEFAULT`

so no matter which external macro is calling, the change will be applied